packages/steppers/src/elements/Stepper.tsx

```diff
@@ -0,0 +1,72 @@
+/**
+ * Copyright Zendesk, Inc.
+ *
+ * Use of this source code is governed under the Apache License, Version 2.0
+ * found at http://www.apache.org/licenses/LICENSE-2.0.
+ */
+
+import React, {
+  useRef,
+  forwardRef,
+  HTMLAttributes,
+  ForwardRefExoticComponent,
+  PropsWithoutRef,
+  RefAttributes
+} from 'react';
+import { StyledStepper } from '../styled';
+import { StepperContext } from '../utils';
+import { Step } from './components/Step';
+import { StepLabel } from './components/StepLabel';
+import { StepContent } from './components/StepContent';
+
+interface IStaticStepperExport<T, P>
+  extends ForwardRefExoticComponent<PropsWithoutRef<P> & RefAttributes<T>> {
+  Step: typeof Step;
+  Label: typeof StepLabel;
+  Content: typeof StepContent;
+}
+
+interface IStepperProps extends HTMLAttributes<HTMLDivElement> {
+  /** Used to show the current step and compute completed steps */
+  activeIndex: number;
+  /** Displays the Stepper in a horizontal layout */
+  isHorizontal: boolean;
+}
+
+/**
+ * Accepts all `<div>` attributes and events. Also accepts static properties:
+
+ *  - `Stepper.Step`
+ *  - `Stepper.StepLabel`
+ *  - `Stepper.StepContent`
+ */
+// eslint-disable-next-line react/display-name
+export const Stepper = forwardRef<HTMLDivElement, IStepperProps>(
+  ({ isHorizontal, activeIndex, ...props }, ref) => {
+    const currentIndexRef = useRef(0);
+    const stepperContext = { isHorizontal, activeIndex, currentIndexRef };
+
+    React.useEffect(() => {
+      return () => {
+        currentIndexRef.current = 0;
+      };
+    }, [props.children]);
```

Comment:
This effect is only used to perform clean up — it resets the current index so that each `Step` can compute the correct step when a `Stepper` has dynamic steps (e.g. the `basic` example which can dynamically render 3, 5, or 10 steps). Without this cleanup, dynamically changing the number of steps causes `Step` to have incorrect calculations. 

This cleanup should not result in a performance hit. It runs once when the component mounts, unmounts, and each time the `activeIndex` changes (`props.children` change).